blog/2019-03-13-KIEServerOptaPlannerTaskAssignment.adoc

```diff
@@ -0,0 +1,228 @@
+        ...
+        scoreDirector.beforeProblemPropertyChanged(toBePinnedTask);
+        toBePinnedTask.setPinned(true);
+        scoreDirector.afterProblemPropertyChanged(toBePinnedTask);
+        ...
+
+        scoreDirector.triggerVariableListeners();
+    }
+}
+----
+
+The full class can be found https://github.com/kiegroup/kie-server-optaplanner-task-assignment/blob/master/task-assignment-kjar/src/main/java/org/kie/server/examples/optaplanner/taskassignment/kjar/domain/realtime/PinTaskProblemFactChange.java[on github].
+
+===  https://docs.optaplanner.org/latestFinal/optaplanner-docs/html_single/index.html#solverConfiguration[Solver Configuration]
+
+The solver configuration file determines how the solving process works. For a KJAR deployed to a KIE Server and since we are using
+Drools for score calculation, use a `ksessionName`. This tells the KieContainer where to find the DRL file.
+Add an optataskKsession to `META-INF/kmodule.xml` file:
+
+[source,xml,options="nowrap"]
+----
+<kbase name="optataskKBase" packages="PATH_TO_SOLVER_RESOURCES">
+    <ksession name="optataskKsession"/>
+</kbase>
+----
+
+And to solver config file:
+
+[source,xml,options="nowrap"]
+----
+<scoreDirectorFactory>
+    <ksessionName>optataskKsession</ksessionName>
+</scoreDirectorFactory>
+----
+
+Notice that both solver configuration and DRL files need to be in the `resources` folder under the path
+`PATH_TO_SOLVER_RESOURCES`.
+
+For real-time planning, set the solver in
+https://docs.optaplanner.org/latestFinal/optaplanner-docs/html_single/index.html#daemon[daemon mode] in order to resume
+solving once a problem fact change is added. This is accomplished by adding the following to the solver config file:
+
+[source,xml,options="nowrap"]
+----
+<daemon>true</daemon>
+----
+
+=== Persistence
+
+Since you will be sending and receiving data to/from KIE Server through REST API, you need to tell it how to marshall/unmarshall
+this data. Read how OptaPlanner
+https://docs.optaplanner.org/latestFinal/optaplanner-docs/html_single/index.html#integrationWithXStream[marshals a score] using Xstream.
+
+OptaPlanner stores all planning entities as objects and references to these objects, which might result in a lot of redundant data
+received from the server. In our example, we are using
+https://github.com/FasterXML/jackson-annotations/wiki/Jackson-Annotations#object-references-identity[@JsonIdentityInfo]
+to avoid such redundancies.
+
+=== Dependencies
+
+The minimum required dependencies to build the OptaPlanner service are: optaplanner-core, optaplanner-persistence-xstream.
+
+NOTE: Because you will be using several OptaPlanner modules, it’s recommended to import the optaplanner-bom
+in Maven’s `dependencyManagement` so that OptaPlanner version is specified only once.
+
+
+== Build and Deployment
+
+Now that you have the KJAR ready, you can deploy it to the KIE Server and perform all planning operations.
+You can send HTTP requests to the server using REST API.
```

Comment:
3x more cases of "you can" that look like unneeded. Non-blocking, so I am going to go ahead and merge and publish anyway, to fix these use a separate PR.